src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python
+
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import argparse
+import collections
+import random
+import math
+import subprocess
+
+# Local packages
+import settings
+path = os.path.join(settings.TEMPLATE_DIR)
+sys.path.append(path)
+import validate # Function
+from constants import _constants # Constant values
+from parse_args import parse_args # Argument parser
+from wite_file import _write_file
+
+_GENERATED_FILENAME = '''binary-ops-{NUMBER}.js''' # The file name prefix of the generated tests
+_BINARY_OPERATORS = ["add", "sub", "div", "mul", "mod", "xor", "and", "or", "rightshift", "leftshift",
+"unsignedrightshift"] # The binary operators used in this script
+
+# Needed for unsigned right shift
+def unsigned32(signed):
```

Comment:
Until there is no common util.py for these kind of functions, you should pack them to the main class.